src/leapfrogai_api/backend/rag/index.py

```
@@ -0,0 +1,113 @@
+
+        if not file_object:
+            raise ValueError("File not found")
+        file_bytes = await crud_file_bucket.download(id_=file_id)
+
+        with tempfile.NamedTemporaryFile(suffix=file_object.filename) as temp_file:
+            temp_file.write(file_bytes)
+            temp_file.seek(0)
+            documents = await load_file(temp_file.name)
+            chunks = await split(documents)
+
+            if len(chunks) == 0:
+                vector_store_file = VectorStoreFile(
+                    id=file_id,
+                    created_at=0,
+                    last_error=LastError(
+                        message="No text found in file", code="parsing_error"
+                    ),
+                    object="vector_store.file",
+                    status=VectorStoreFileStatus.FAILED.value,
+                    vector_store_id=vector_store_id,
+                )
+                return await crud_vector_store_file.create(object_=vector_store_file)
+
+            vector_store_file = VectorStoreFile(
+                id=file_id,
+                created_at=0,
+                last_error=None,
+                object="vector_store.file",
+                status=VectorStoreFileStatus.IN_PROGRESS.value,
+                vector_store_id=vector_store_id,
+            )
+
+            vector_store_file = await crud_vector_store_file.create(
+                object_=vector_store_file
+            )
+
+            try:
+                vector_store_client = AsyncSupabaseVectorStore(
+                    db=self.db, embedding=self.embeddings
+                )
+
+                ids = await vector_store_client.aadd_documents(
+                    documents=chunks,
+                    vector_store_id=vector_store_id,
+                    file_id=file_id,
+                )
+
+                if len(ids) == 0:
+                    vector_store_file.status = VectorStoreFileStatus.FAILED.value
```

Should we raise an exception here as well?